src/app/doctor/doctor.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "../person/person.hpp"
+
+#include <set>
+
+class Visit;
+
+class Doctor : public Person, public std::enable_shared_from_this<Doctor>
+{
+    std::set<std::shared_ptr<Visit>> visitAssociation_;
```

Comment:
Why set was used here? 

Please correct field name to snake case.
I also propse to simply name it `visits_` instead `visit_association_`